docs/project/list-of-diagnostics.md

```diff
@@ -1,15 +1,44 @@
+The diagnostic id values reserved for obsoletions are `SYSLIB0001` through `SYSLIB0999`. When obsoleting an API, claim the next three-digit identifier in the `SYSLIB0###` sequence and add it to the list below. The URL template for all obsoletions is `https://aka.ms/dotnet-warnings/{0}`. The `{0}` placeholder is replaced by the compiler with the `SYSLIB0###` identifier.
 
-Currently the identifiers `SYSLIB0001` through `SYSLIB0999` are carved out for obsoletions. If we wish to introduce analyzer warnings not related to obsoletion in the future, we should begin at a different range, such as `SYSLIB2000`.
+The acceptance criteria for adding an obsoletion includes:
 
-## Current obsoletions (`SYSLIB0001` - `SYSLIB0999`)
+* Add the obsoletion to the table below, claiming the next diagnostic id
+    * Ensure the description is meaningful within the context of this table, and without requiring the context of the calling code
+* Add new constants to `src\libraries\Common\src\System\Obsoletions.cs`, following the existing conventions
+    * A `...Message` const using the same description added to the table below
+    * A `...DiagId` const for the `SYSLIB0###` id
+* Annotate `src` files by referring to the constants defined from `Obsoletions.cs`
+    * Specify the `UrlFormat = Obsoletions.SharedUrlFormat`
+    * Example: `[Obsolete(Obsoletions.CodeAccessSecurityMessage, DiagnosticId = Obsoletions.CodeAccessSecurityDiagId, UrlFormat = Obsoletions.SharedUrlFormat)]`
+    * If the `Obsoletions` type is not available in the project, link it into the project
+        * `<Compile Include="$(CommonPath)System\Obsoletions.cs" Link="Common\System\Obsoletions.cs" />`
+* Annotate `ref` files using the hard-coded strings copied from `Obsoletions.cs`
+    * This matches our general pattern of `ref` files using hard-coded attribute strings
+    * Example: `[System.ObsoleteAttribute("The UTF-7 encoding is insecure and should not be used. Consider using UTF-8 instead.", DiagnosticId = "SYSLIB0001", UrlFormat = "https://aka.ms/dotnet-warnings/{0}")]`
+* If the library builds against downlevel targets earlier than .NET 5.0, then add an internal copy of `ObsoleteAttribute`
+    * The compiler recognizes internal implementations of `ObsoleteAttribute` to enable the `DiagnosticId` and `UrlFormat` properties to light up downlevel
+    * An MSBuild property can be added to the project's first `<PropertyGroup>` to achieve this easily
+    * Example: `<IncludeInternalObsoleteAttribute>true</IncludeInternalObsoleteAttribute>`
+    * This will need to be specified in both the `src` and `ref` projects
+* Apply the `breaking-change` label to the PR that introduces the obsoletion
+    * A bot will automatically apply the `needs-breaking-change-doc-created` label when the `breaking-change` label is detected
+* Follow up with the normal breaking change process to communicate and document the breaking change
+
```

Comment:
Does the aka.ms/dotnet-warnings/xxxx link automatically do something useful after the above steps, or is there another step for that?

Reply:
Oh, good point, @danmoseley. I'll get those instructions added in.